settings.go

```diff
@@ -38,17 +40,73 @@ type Settings struct {
 	Callback string
 }
 
-// LoadSettings extracts the git and gitOpenPull sections from $HOME/.gitconfig and .git/config
-func LoadSettings(ctx context.Context) (*Settings, error) {
+// this function tries to get settings from the environment variables
+func GetEnvSettings() (*Settings, error) {
+	token := os.Getenv("GITOPENPULL_TOKEN") 
+	user := os.Getenv("GITOPENPULL_USER")
+	baseAccount := os.Getenv("GITOPENPULL_BASE_ACCOUNT")
+	baseRepo := os.Getenv("GITOPENPULL_BASE_REPO")
 
-	body, err := RunGit(ctx, "config", "--list")
-	if err != nil {
-		return nil, err
+	if token == "" || user == "" || baseAccount == "" || baseRepo == "" {
+		return nil, errors.New("You must have GITOPENPULL_TOKEN, GITOPENPULL_USER, GITOPENPULL_BASE_ACCOUNT and GITOPENPULL_BASE_REPO set as env variables")
 	}
+
+	s := Settings{
+		BaseBranch: "master",
+		Editor: "/usr/bin/vi",
+		Token: token,
+		User: user,
+		BaseAccount: baseAccount,
+		BaseRepo: baseRepo, 
+		PreProcess: os.Getenv("GITOPENPULL_PRE_PROCESS"),
+		PostProcess: os.Getenv("GITOPENPULL_POST_PROCESS"),
+		Callback: os.Getenv("GITOPENPULL_CALLBACK"),
+	}
+
+	baseBranch := os.Getenv("GITOPENPULL_BASE_BRANCH")
+	if baseBranch != "" {
+		s.BaseBranch = baseBranch
+	}
+	
+	mcmStr := os.Getenv("GITOPENPULL_MAINTAINERS_CAN_MODIFY")
+	mcm := true
+	var err error
+	if mcmStr != "" {
+		mcm, err = strconv.ParseBool(mcmStr)
+		if err != nil {
+			return nil, err
+		}
+	} 
+	s.MaintainersCanModify = mcm
+
+	editor := os.Getenv("GITOPENPULL_EDITOR")
+	if editor != "" {
+		s.Editor = editor
+	}
+	
+	return &s, nil
+}
+
+// LoadSettings extracts the git and gitOpenPull sections from $HOME/.gitconfig and .git/config
+func LoadSettings(ctx context.Context, settingsFromEnv bool) (*Settings, error) {
+	if settingsFromEnv {
```

Comment:
I think it's more common to check for environment variables last (after the scan loop) and allow any that are set to take priority (Instead of having a command line flag use environment variables). 

That allows 0 or more settings to be overridden by environment variables without requiring them all. 

What do you think of moving this to after the scan loop as a `LoadSettingsFromEnv(s *Settings)` function?

Reply:
works for me! will update 👍